app/services/candidate_interface/validate_course_query_string_params.rb

```diff
@@ -0,0 +1,46 @@
+module CandidateInterface
+  class ValidateCourseQueryStringParams
+    attr_accessor :return_course
+
+    def initialize(provider_code:, course_code:)
+      @provider_code = provider_code
+      @course_code = course_code
+      @can_apply_on_apply = false
+      @course_on_find = false
+      @return_course = nil
+    end
+
+    def execute
+      begin
```

Comment:
You can skip the `begin` and `end` here if you move the rescue block to the end of the method definition.